dts/bindings/sensor/st,lis2mdl-i2c.yaml

```diff
@@ -17,3 +17,17 @@ properties:
         This pin defaults to active high when produced by the sensor.
         The property value should ensure the flags properly describe
         the signal that is presented to the driver.
+
+    single-mode:
+      type: boolean
+      required: false
```

Comment:
Yes, If I change it to true, then it is required to be set in .dts file. So it will always be enabled which is not my intention. 

If the user obviates it, the driver instance will work in continuous mode.

Reply:
if removed from bindings, it should take `false` by default

Reply:
If the user does not define it in the .dts file, it will be considered as false (0) by default. 

https://docs.zephyrproject.org/latest/guides/dts/bindings.html#bindings-file-syntax
#   - 'type: boolean' is for properties used as flags that don't take a value,
#     like
#
#         hw-flow-control;
#
#     The macro generated for the property gets set to 1 if the property exists
#     on the node, and to 0 otherwise. When combined with 'required: true',
#     this type just forces the flag to appear on the node. The output will
#     always be 1 in that case.